dhis-2/dhis-support/dhis-support-commons/src/main/java/org/hisp/dhis/commons/jackson/config/geometry/GeometrySerializer.java

```diff
@@ -0,0 +1,249 @@
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.commons.jackson.config.geometry;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.geom.GeometryCollection;
+import org.locationtech.jts.geom.LineString;
+import org.locationtech.jts.geom.MultiLineString;
+import org.locationtech.jts.geom.MultiPoint;
+import org.locationtech.jts.geom.MultiPolygon;
+import org.locationtech.jts.geom.Point;
+import org.locationtech.jts.geom.Polygon;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.RuntimeJsonMappingException;
+import com.fasterxml.jackson.databind.SerializerProvider;
+
+/**
+ * A custom serializer for a Geometry
+ */
+/*
+ * A custom serializer is needed as
+ * com.graphhopper.external:jackson-datatype-jts is obsolete and not compatible
+ * with last version of Hibernate library
+ */
+public class GeometrySerializer extends JsonSerializer<Geometry>
+{
+    private static final String COORDINATES = "coordinates";
+
+    public void serialize( Geometry value, JsonGenerator jgen, SerializerProvider provider )
+        throws IOException
+    {
+        this.writeGeometry( jgen, value );
+    }
+
+    public void writeGeometry( JsonGenerator jgen, Geometry value )
+        throws IOException
+    {
+        if ( value instanceof Polygon )
+        {
+            this.writePolygon( jgen, (Polygon) value );
+        }
+        else if ( value instanceof Point )
+        {
+            this.writePoint( jgen, (Point) value );
+        }
+        else if ( value instanceof MultiPoint )
+        {
+            this.writeMultiPoint( jgen, (MultiPoint) value );
+        }
+        else if ( value instanceof MultiPolygon )
+        {
+            this.writeMultiPolygon( jgen, (MultiPolygon) value );
+        }
+        else if ( value instanceof LineString )
+        {
+            this.writeLineString( jgen, (LineString) value );
+        }
+        else if ( value instanceof MultiLineString )
+        {
+            this.writeMultiLineString( jgen, (MultiLineString) value );
+        }
+        else
+        {
+            if ( !(value instanceof GeometryCollection) )
+            {
+                throw new RuntimeJsonMappingException( "Geometry type " + value.getClass().getName()
+                    + " cannot be serialized as GeoJSON.Supported types are: "
+                    + Arrays.asList( Point.class.getName(), LineString.class.getName(), Polygon.class.getName(),
+                        MultiPoint.class.getName(), MultiLineString.class.getName(), MultiPolygon.class.getName(),
+                        GeometryCollection.class.getName() ) );
+            }
+
+            this.writeGeometryCollection( jgen, (GeometryCollection) value );
```

Comment:
can we put this in its own else if statement. then throw the RuntimeJsonMappingException outside of the if/else if statement

Reply:
I rather not touch this class.
I think next step should be try to send a PR to the library to see if they upgrade the jts-core version or maybe fork it and create our own version of jackson-datatype-jts.